src/mlpack/methods/ann/activation_functions/hard_sigmoid_function.hpp

```diff
@@ -0,0 +1,104 @@
+/**
+ * @file hard_sigmoid_function.hpp
+ * @author Bishwa Karki
+ *
+ * Definition and implementation of the hard sigmoid function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_HARD_SIGMOID_FUNCTION_HPP
+#define MLPACK_METHODS_ANN_ACTIVATION_FUNCTIONS_HARD_SIGMOID_FUNCTION_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <algorithm>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The hard sigmoid function, defined by
+ *
+ * @f{eqnarray*}{
+ * if x < -2.5 \\
+ *    @return 0
+ * elseif x > 2.5 \\
+ *    @return 1
+ * else \\
+ *    @return x * .2 + .5
+ * endif
+ * if x > 2.5 || x < -2.5
+ *    f'(x) &=& 0 \\
+ * else
+ *    f'(x) &=& 0.2
+ * @f}
+ */
+class HardSigmoidFunction
+{
+ public:
+  /**
+   * @Computes the hard sigmoid function.
+   * 
+   * @param x 
+   * @return double 
+   */
+  static double Fn(const double x)
+  {
+    return std::min(1.0, std::max(0.0, 0.2 * x + 0.5));
+  }
+
+  /**
+   * Computes the hard sigmoid function.
+   *
+   * @tparam InputVecType
+   * @tparam OutputVecType
+   * @param x
+   * @param y
+   */
+  template<typename InputVecType, typename OutputVecType>
+  static void Fn(const InputVecType& x, OutputVecType& y)
+  {
+    y = x;
```

Comment:
I think we just need to use `resize()` here.

Reply:
Better to use ``set_size``, since it should be much faster.